requirements/prod.in

```diff
@@ -50,7 +50,6 @@ rich-text-renderer==0.2.8
 sentry-processor==0.0.1
 sentry-sdk==2.29.1
 supervisor==4.2.5
-typing-extensions==4.13.2  # Temporarily here for prod <-> dev compatibility. Selenium is pinned with ~4.13.2 but running without selenium chooses ==4.14 which then fails when building dev requirements.
```

Comment:
This is the main change, prod now not conflicting with what the latest selenium version used resolves with in dev.

This was not pinned before this temp addition, so seems safe and appropriate to unpin back now, when the clash no longer happens. Quite the contrary, this hard pin now keeps blocking upgrades to newer selenium versions, which might be necessary to retain compatibility with the rest of the deps (esp. the recent sec bump of urllib/requests…)